### DISCUSSION TITLE
How to use LoadingIndicator in Switch on_switch_changed callback?

### DISCUSSION THREAD

**Discussion by Da-Buche** [Textualize/textual#2845] [Q&A] [answered]
Hi,

I have a switch that triggers a server when toggled. 
It can takes about ten seconds so I wanted to display a loading indicator until the server answers.

Unfortunately, the screen is only refreshed once the on_switch_changed call is done so the loading indicator is never displayed

Can someone help me achieve what I want ?

Cheers
Aurélien

**Comment by davep** [comment, marked as answer]
By the sounds of it you need to be looking at the [Worker API](https://textual.textualize.io/guide/workers/).

**Reply by Da-Buche** [reply]
Hi Dave,

Sorry for the late response, I didn't get much time to work on my textual interface project since

I made a quick trial with the worker API.
I couldn't achieve what I wanted yet but it seems to be the proper solution.
I'll take a deeper look as soon as possible!

Thanks for your answer.

Cheers,
Aurélien
